base/libgit2/libgit2.jl

```diff
@@ -196,13 +196,21 @@ function make_payload{P<:AbstractCredentials}(payload::Nullable{P})
 end
 
 """
-    fetch(repo::GitRepo; remote::AbstractString="origin", remoteurl::AbstractString="", refspecs=AbstractString[], payload=Nullable{AbstractCredentials}())
+    fetch(repo::GitRepo; kwargs...)
+
+Fetches updates from an upstream of the repository `repo`.
+
+The keyword arguments are:
+  * `remote::AbstractString="origin"`: which remote, specified by name,
+    of `repo` to fetch from. If this is empty, the URL will be used to
+    construct an anonymous remote.
+  * `remoteurl::AbstractString=""`: the URL of `remote`. If not specified,
+    will be assumed based on the given name of `remote`.
```

Comment:
it may be worth explaining in what way this assumption is made, but can probably be a future pr